pkg/publish/openshift.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2021 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package publish
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/ko/pkg/build"
+)
+
+const (
+	// OpenshiftDomain is a sentinel "registry" that represents loading images into
+	// Openshift's internal registry.
+	OpenshiftDomain = "ocp.local"
+
+	// This line is printed by the tunnel command when it tells us which port it uses.
+	portPrefix = "Forwarding from 127.0.0.1:"
+)
+
+type ocpPublisher struct {
+	inner  Interface
+	tunnel *os.Process
+}
+
+// NewOpenshiftPublisher returns a new publish.Interface that loads images into
+// Openshift's internal registry.
+func NewOpenshiftPublisher(namer Namer, tags []string) (Interface, error) {
```

Comment:
supernit: I think the trademark is technically "OpenShift" with a capital S, I'm ashamed to even be this pedantic.

Reply:
ack!